src/commands/registries/azure/DockerSiteCreateStep.ts

```diff
@@ -54,33 +57,76 @@ export class DockerSiteCreateStep extends AzureWizardExecuteStep<IAppServiceCont
         context.site = await client.webApps.createOrUpdate(nonNullValueAndProp(context.resourceGroup, 'name'), nonNullProp(context, 'newSiteName'), siteEnvelope);
     }
 
-    private async getSiteConfig(context: IAppServiceContainerWizardContext): Promise<WebSiteManagementModels.SiteConfig> {
-        // Temporary workaround until Arc adds support for managed identity, so use usename and password instead.
-        // When customLocation is set, then user is deploying to Arc.
-        const registryTreeItem: RegistryTreeItemBase = this.node.parent.parent;
-        if (registryTreeItem instanceof AzureRegistryTreeItem && context.customLocation) {
-            const appSettings: WebSiteManagementModels.NameValuePair[] = [];
-            const cred = await registryTreeItem.tryGetAdminCredentials(context);
-            if (!cred) {
+    private async getNewSiteConfig(context: IAppServiceContainerWizardContext): Promise<WebSiteManagementModels.SiteConfig> {
+        const registryTI: RegistryTreeItemBase = this.node.parent.parent;
+
+        let username: string | undefined;
+        let password: string | undefined;
+        let registryUrl: string | undefined;
+        const appSettings: WebSiteManagementModels.NameValuePair[] = [];
+
+        // Scenarios:
+        // ACR -> App Service, NOT Arc App Service. Use managed service identity.
+        if (registryTI instanceof AzureRegistryTreeItem && !context.customLocation) {
+            appSettings.push({ name: 'DOCKER_ENABLE_CI', value: 'true' });
+
+            // Don't need an image, username, or password--just create an empty web app to assign permissions and then configure with an image
+            // `DockerAssignAcrPullRoleStep` handles it after that
+            return {
+                acrUseManagedIdentityCreds: true,
+                appSettings
+            };
+        }
+        // ACR -> Arc App Service. Use regular auth. Same as any V2 registry but different way of getting auth.
+        else if (registryTI instanceof AzureRegistryTreeItem && context.customLocation) {
+            const cred = await registryTI.tryGetAdminCredentials(context);
+            if (!cred?.username || !cred?.passwords?.[0]?.value) {
                 throw new Error(localize('vscode-docker.commands.registries.azure.dockersitecreatestep.notAdminEnabled', 'Azure App service deployment on Azure Arc only supports running images from Azure Container Registries with admin enabled'));

```

Comment:
😞 Are we aware of any plans for ACR + Arc App Service to avoid admin credential requirement?

Reply:
Yes, in fact I should check with them again. Last I heard they needed changes from the identity team to support MSI but it was not possible at the time (before my ICL so several months ago).

Reply:
Current ETA: "it's going to be a while".

Reply:
😑